src/adapter/src/coord/timestamp_oracle/metrics.rs

```diff
@@ -138,6 +147,13 @@ impl MetricsVecs {
                 help: "time spent in retry loop backoff",
                 var_labels: ["timeline", "op"],
             )),
+
+            batched_op_count: registry.register(metric!(
+                name: "mz_ts_oracle_batched_op_count",
+                help: "number of operations that were batched into one external operation",
+                var_labels: ["timeline", "op"],
+                buckets: HISTOGRAM_COUNT_BUCKETS.to_vec(),
```

Comment:
I picked a histogram because we'll only have one instance: there's one timeline and one op that does batching.

And I didn't come up with other ways of recording how well batching works. The one thing I considered is two metrics, roughly: `ops_count` and `batches_count`. And then you could get `ops_count / batches_count` as your "how good is my batching on average". But on a mostly idle env that would be strongly dragged towards `1` because at idle not much batching is happening. So you don't see spikes in batching well, for example when running a benchmark or just a large enough parallel workload.